### PR TITLE
Adding SearchClientBuilderExtensions

### LINKED ISSUE #10848: [Search] Adding ClientBuilderExtensions
Add `SearchClientBuilderExtensions` to `Azure.Search.Documents`.

### PR DESCRIPTION
**Pull request by tg-msft**
Fixes #10848

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/SearchClientBuilderExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        /// </param>
+        /// <param name="credential">
+        /// Required.  The API key credential used to authenticate requests
+        /// against the search service.  You need to use an admin key to
+        /// modify the documents in a Search Index.  See
+        /// <see href="https://docs.microsoft.com/azure/search/search-security-api-keys"/>
+        /// for more information about API keys in Azure Cognitive Search.
+        /// </param>
+        /// <returns>An Azure client builder.</returns>
+        public static IAzureClientBuilder<SearchIndexClient, SearchClientOptions> AddSearchIndexClient<TBuilder>(
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/SearchClientBuilderExtensions.cs:86]: Thanks for adding this. It slipped my mind. Should we add one for `SearchIndexerClient` as well?

**Review by heaths — APPROVED** [review]

**Comment by heaths** [comment]
Should also add something to the CHANGELOG.md.

**Comment by tg-msft** [comment]
I updated the changelog and put the release date in there as well for good measure.

**Comment by tg-msft** [comment]
_(Merging to keep things moving, but happy to continue the discussion about `SearchIndexerClient` and add that as separate PR if we decide to.)_

File: sdk/search/Azure.Search.Documents/src/SearchClientBuilderExtensions.cs (hunk 1 above)
Comment (heaths) [reply sdk/search/Azure.Search.Documents/src/SearchClientBuilderExtensions.cs:86]: No, I think you're explanation makes sense. I was leaning that way as well initially (and should've mentioned that).

**Pull request merged**